### PR TITLE
Fix #7036: Clean up file uploads orphaned by WebKit on app launch

### LINKED ISSUE #7036: [WebKit] File uploaded are not cleaned up when Brave is force closed with the page open
### Description: 

When a user uploads a file through a web page, WebKit will copy said file to the `/tmp` directory in the app sandbox. If the user does not close the tab they used to upload (thus destroying the WKWebView) before the app terminates, then WebKit no longer cleans up the file and it remains on disk until the OS decides to purge `/tmp` directories to recover disk space.

Related to https://github.com/brave/brave-ios/issues/6822

### Steps to Reproduce 
  1. Visit a page where you can upload files to (e.g. dropbox, etc.)
  2. Select a file through the relevant file picker (depending on the website, could be a photo picker, video picker, files picker). You don't actually have to upload it, just select it through the form
  3. Force close the app (via app tray)

### Actual result:
In the `Sandbox Inspector`, a folder prefixed with `WKVideoUpload`, `WKWebFileUpload`, or `WKFileUploadPanel` will be present in the `tmp` root folder. Inside that folder will be the file you selected in step 2

### Expected result:
In the `Sandbox Inspector` there should be no leftover data from your file uploads in prior browsing sessions when launching the app

### Reproduces how often:
Easily

### Brave Version: 
Probably for a while, this is a WebKit bug

### PR DESCRIPTION
**Pull request by kylehickinson**
## Summary of Changes

This pull request fixes #7036 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan
See issue

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: App/iOS/Delegates/AppDelegate.swift [hunk 1]
```diff
@@ -469,6 +474,34 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+  private nonisolated func cleanUpWebKitPendingFileUploads() async {
+    // When a user uploads a file through a web page, WebKit will copy said file to the `/tmp` directory in
+    // the app sandbox. If the user does not close the tab they used to upload (thus destroying the WKWebView)
+    // before the app terminates, then WebKit no longer cleans up the file and it remains on disk until
+    // the OS decides to purge /tmp directories to recover disk space. This function will attempt to remove
+    // known temporary folders that WebKit creates to clean up any files that may be lingering there due to
+    // this WebKit bug.
+    let fileManager = FileManager.default
+    let tmp = fileManager.temporaryDirectory
+    let knownFolders: Set<String> = ["WKWebFileUpload", "WKFileUploadPanel", "WKVideoUpload"]
```
Comment (soner-yuksel) [inline App/iOS/Delegates/AppDelegate.swift:487]: @kylehickinson Are these all in the WKFileUploadPanel form?

Comment (kylehickinson) [reply App/iOS/Delegates/AppDelegate.swift:487]: I know for sure that `WKFileUploadPanel` is created in root folder, but based on user reports I am guessing the other 2 are also created in some form as well. Perhaps from older iOS versions? Or specific types of file pickers on websites?

Comment (kylehickinson) [reply App/iOS/Delegates/AppDelegate.swift:487]: This method at least describes `WKVideoUpload` https://github.com/WebKit/WebKit/blob/0c6de669a59bddb53aa318ddf458c700336c6141/Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm#L354

Comment (soner-yuksel) [reply App/iOS/Delegates/AppDelegate.swift:487]: Sounds perfect for this particular use-case.

**Review by soner-yuksel — APPROVED** [review]

File: App/iOS/Delegates/AppDelegate.swift (hunk 1 above)
Comment (kylehickinson) [reply App/iOS/Delegates/AppDelegate.swift:487]: And this one `WKWebFileUpload`: https://github.com/WebKit/WebKit/blob/0c6de669a59bddb53aa318ddf458c700336c6141/Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm#L1144

**Pull request merged**